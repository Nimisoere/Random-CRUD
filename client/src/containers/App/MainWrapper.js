import React, { memo } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { ThemeProps } from "../../shared/prop-types/ReducerProps"

export const MainWrapper = memo(props => {
  const { theme, children } = props;

  return (
    <div className={theme.className}>
      <div className="wrapper">{children}</div>
    </div>
  );
});

MainWrapper.propTypes = {
  theme: ThemeProps.isRequired,
  children: PropTypes.element.isRequired
};

export default connect(state => ({
  theme: state.theme
}))(MainWrapper);
