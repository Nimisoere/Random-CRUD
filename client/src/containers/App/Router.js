import React from "react";
import { Route, Switch } from "react-router-dom";
import Layout from "../Layout/index";
import MainWrapper from "../App/MainWrapper";

import Dashboard from "../Dashboard/Router";
import Shipments from "../Shipments/Router"

const Router = () => (
  <Switch>
    <MainWrapper>
      <main>
        <div>
          <Layout />
          <div className="container__wrap">
            <Route exact path="/" component={Dashboard} />
            <Route path="/shipments" component={Shipments} />
          </div>
        </div>
      </main>
    </MainWrapper>
  </Switch>
);

export default Router;
