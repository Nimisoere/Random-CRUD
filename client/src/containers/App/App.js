import React, { useState, Fragment, useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import { hot } from 'react-hot-loader';

import Router from "./Router";

import ScrollToTop from "./ScrollToTop";

import "bootstrap/dist/css/bootstrap.css";
import "../../styles/app.scss";

const App = () => {
  const [loading, setLoading] = useState(true);
  const [loaded, setLoaded] = useState(false);

  const loadPage = () => {
    window.addEventListener("load", () => {
      setLoading({ loading: false });
      setTimeout(() => setLoaded({ loaded: true }), 200);
    });
  };

  useEffect(() => {
    loadPage();
  }, []);

  return (
    <BrowserRouter>
      <ScrollToTop>
        <Fragment>
          {!loaded && (
            <div className={`load${loading ? "" : " loaded"}`}>
              <div className="load__icon-wrap">
                <svg className="load__icon">
                  <path
                    fill="#003553"
                    d="M12,4V2A10,10 0 0,0 2,12H4A8,8 0 0,1 12,4Z"
                  />
                </svg>
              </div>
            </div>
          )}
          <div>
            <Router />
          </div>
        </Fragment>
      </ScrollToTop>
    </BrowserRouter>
  );
};

export default hot(module)(App);
