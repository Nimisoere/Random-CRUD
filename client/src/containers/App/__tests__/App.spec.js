import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "../App";

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const enzymeWrapper = shallow(<App />);
  return {
    enzymeWrapper
  };
}

describe("components", () => {
  describe("App", () => {
    it("Match Snapshot", () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
