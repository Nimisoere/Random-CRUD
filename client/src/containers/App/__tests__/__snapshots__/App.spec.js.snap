// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components App Match Snapshot 1`] = `ShallowWrapper {}`;
